lib/pbench/cli/server/reindex.py

```diff
@@ -0,0 +1,318 @@
+            )
+
+
+@click.command(name="pbench-reindex")
+@pass_cli_context
+@click.option(
+    "--delete", default=False, is_flag=True, help="Delete index for selected dataset(s)"
+)
+@click.option(
+    "--detail",
+    default=False,
+    is_flag=True,
+    help="Provide extra diagnostic information",
+)
+@click.option(
+    "--errors",
+    default=False,
+    is_flag=True,
+    help="Show individual dataset errors",
+)
+@click.option("--id", type=str, multiple=True, help="Select dataset by resource ID")
+@click.option(
+    "--indexing",
+    type=click.Choice(["enable", "disable"], case_sensitive=False),
+    help="Enable or disable the Pbench Server indexer for future uploads",
+)
+@click.option(
+    "--list", default=False, is_flag=True, help="Show dataset indexing status"
+)
+@click.option("--name", type=str, multiple=True, help="Select dataset by name")
+@click.option(
+    "--progress", type=float, default=0.0, help="Show periodic progress messages"
+)
+@click.option(
+    "--reindex", is_flag=True, default=False, help="Reindex selected datasets"
+)
+@click.option(
+    "--since",
+    type=DateParser(),
+    help="Select datasets uploaded on or after specified date/time",
+)
+@click.option(
+    "--until",
+    type=DateParser(),
+    help="Select datasets uploaded on or before specified date/time",
+)
+@click.option(
+    "--verify", "-v", default=False, is_flag=True, help="Display detailed messages"
+)
+@common_options
+def reindex(context: object, **kwargs):
+    """Control dataset indexing.
+
+    This can globally enable or disable indexing for the server when datasets
+    are uploaded.
+
+    It can also report on the indexing status of datasets, as well as deleting
+    a dataset's indexed documents and (re-)indexing datasets.
+    \f
+
+    Args:
+        context: click context
+        kwargs: click options
+    """
+    global detailer, verifier, watcher
+    detailer = Detail(kwargs.get("detail"), kwargs.get("errors"))
+    verifier = Verify(kwargs.get("verify"))
+    watcher = Watch(kwargs.get("progress"))
+
+    try:
+        config = config_setup(context)
+        logger = get_pbench_logger("pbench-reindex", config)
+        es_url = config.get("Indexing", "uri")
+        ca_bundle = config.get("Indexing", "ca_bundle")
+        kwargs["_es"] = (es_url, ca_bundle)
+        kwargs["_logger"] = logger
+
+        # Check whether to enable or disable automatic indexing on upload.
+        indexing = kwargs.get("indexing")
+        if indexing:
+            state = indexing == "enable"
+            detailer.message(f"{indexing} upload indexing")
+            ServerSetting.set(key=OPTION_SERVER_INDEXING, value=state)
+
+        # Operate on individual datasets if selected
+        if (SELECTORS | OPERATORS) & set(k for k, v in kwargs.items() if v):
+            verifier.status("updating selected datasets")
+            worker(kwargs)
+        else:
+            click.echo("nothing to do", err=True)
```

Comment:
Should we be invoking `detailer.error()` or `detailer.warning()` instead of calling `click.echo()` directly, here?

Similarly at line 315?  (Perhaps you want that one to be unconditional.)

Reply:
This is definitely deliberately unconditional, although I did debate that a moment or two. The alternative is to return successfully with no output. OK, that implies, perhaps, that nothing's selected, but I felt I wanted to be a bit more explicit.